checkov/common/util/json_utils.py

```diff
@@ -30,9 +34,62 @@ def default(self, o: Any) -> Any:
             return o.__dict__
         elif isinstance(o, type_of_function):
             return str(o)
+        elif isinstance(o, TFDefinitionKey):
+            return str(o)
+        elif isinstance(o, TFModule):
+            return str(o)
         else:
             return json.JSONEncoder.default(self, o)
 
+    def encode(self, obj):
+        return super().encode(self._encode(obj))
+
+    def _encode(self, obj):
+        if isinstance(obj, dict):
+            return {self.encode_key(k): v for k, v in obj.items()}
+        else:
+            return obj
+
+    @staticmethod
+    def encode_key(key):
+        if isinstance(key, TFDefinitionKey):
+            return str(key)
+        if isinstance(key, TFModule):
+            return str(key)
+        else:
+            return key
+
+
+def object_hook(dct):
```

Comment:
can you rename this function to something a bit more clear? (or explain it to me if I'm missing something)

Reply:
object_hook is the standard way to pass a custom decoder to `json.loads`